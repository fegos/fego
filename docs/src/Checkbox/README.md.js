webpackJsonp([14],{464:function(a,n){a.exports={content:["article",["h1","\u590d\u9009\u6846 Checkbox"],["h3","\u7ec4\u4ef6\u63cf\u8ff0"],["ul",["li",["p","\u540c\u4e8e\u5411\u7528\u6237\u63d0\u4f9b\u591a\u9009\u9009\u62e9\u3002"]]],["h3","\u793a\u4f8b\u4ee3\u7801"],["pre",{lang:"html",highlighted:'<span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>Checkbox</span> <span class="token attr-name">defaultChecked</span><span class="token punctuation">></span></span>\u9ed8\u8ba4\u9009\u4e2d\u7684\u590d\u9009\u6846<span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>Checkbox</span><span class="token punctuation">></span></span>\n\n<span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>Checkbox.Group</span> <span class="token attr-name">name</span><span class="token attr-value"><span class="token punctuation">=</span><span class="token punctuation">"</span>cbox<span class="token punctuation">"</span></span><span class="token punctuation">></span></span>\n  <span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>Checkbox</span> <span class="token attr-name">value</span><span class="token attr-value"><span class="token punctuation">=</span><span class="token punctuation">"</span>chi<span class="token punctuation">"</span></span><span class="token punctuation">></span></span>\u8bed\u6587<span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>Checkbox</span><span class="token punctuation">></span></span>\n  <span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>Checkbox</span> <span class="token attr-name">value</span><span class="token attr-value"><span class="token punctuation">=</span><span class="token punctuation">"</span>mat<span class="token punctuation">"</span></span><span class="token punctuation">></span></span>\u6570\u5b66<span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>Checkbox</span><span class="token punctuation">></span></span>\n  <span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>Checkbox</span> <span class="token attr-name">value</span><span class="token attr-value"><span class="token punctuation">=</span><span class="token punctuation">"</span>eng<span class="token punctuation">"</span></span><span class="token punctuation">></span></span>\u82f1\u8bed<span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>Checkbox</span><span class="token punctuation">></span></span>\n  <span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>Checkbox</span> <span class="token attr-name">value</span><span class="token attr-value"><span class="token punctuation">=</span><span class="token punctuation">"</span>phy<span class="token punctuation">"</span></span><span class="token punctuation">></span></span>\u7269\u7406<span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>Checkbox</span><span class="token punctuation">></span></span>\n  <span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>Checkbox</span> <span class="token attr-name">value</span><span class="token attr-value"><span class="token punctuation">=</span><span class="token punctuation">"</span>che<span class="token punctuation">"</span></span><span class="token punctuation">></span></span>\u5316\u5b66<span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>Checkbox</span><span class="token punctuation">></span></span>\n  <span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>Checkbox</span> <span class="token attr-name">value</span><span class="token attr-value"><span class="token punctuation">=</span><span class="token punctuation">"</span>bio<span class="token punctuation">"</span></span><span class="token punctuation">></span></span>\u751f\u7269<span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>Checkbox</span><span class="token punctuation">></span></span>\n<span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>Checkbox.Group</span><span class="token punctuation">></span></span>'},["code",'<Checkbox defaultChecked>\u9ed8\u8ba4\u9009\u4e2d\u7684\u590d\u9009\u6846</Checkbox>\n\n<Checkbox.Group name="cbox">\n  <Checkbox value="chi">\u8bed\u6587</Checkbox>\n  <Checkbox value="mat">\u6570\u5b66</Checkbox>\n  <Checkbox value="eng">\u82f1\u8bed</Checkbox>\n  <Checkbox value="phy">\u7269\u7406</Checkbox>\n  <Checkbox value="che">\u5316\u5b66</Checkbox>\n  <Checkbox value="bio">\u751f\u7269</Checkbox>\n</Checkbox.Group>']],["h2","API"],["h3","Checkbox props\u5217\u8868"],["table",["thead",["tr",["th","\u5c5e\u6027"],["th","\u8bf4\u660e"],["th","\u7c7b\u578b"],["th","\u9ed8\u8ba4\u503c"]]],["tbody",["tr",["td","label"],["td","\u590d\u9009\u6846\u6587\u6848"],["td","string"],["td"]],["tr",["td","children"],["td","\u590d\u9009\u6846\u6587\u6848\uff0c\u4f7f\u7528\u65f6label\u5c5e\u6027\u5931\u6548"],["td","react node"],["td"]],["tr",["td","checked"],["td","\u662f\u5426\u9009\u4e2d\uff0c\u53d7\u63a7\u5c5e\u6027"],["td","boolean"],["td"]],["tr",["td","defaultChecked"],["td","\u662f\u5426\u9009\u4e2d\uff0c\u975e\u53d7\u63a7\u5c5e\u6027"],["td","boolean"],["td"]],["tr",["td","disabled"],["td","\u662f\u5426\u7981\u7528"],["td","boolean"],["td","false"]],["tr",["td","onChange"],["td","\u53d8\u5316\u56de\u8c03"],["td","function(e){}"],["td"]],["tr",["td","indeterminate"],["td","\u8bbe\u7f6e indeterminate \u72b6\u6001\uff0c\u8d1f\u8d23\u6837\u5f0f\u63a7\u5236"],["td","boolean"],["td","false"]]]],["h3","Checkbox.Group props\u5217\u8868"],["table",["thead",["tr",["th","\u5c5e\u6027"],["th","\u8bf4\u660e"],["th","\u7c7b\u578b"],["th","\u9ed8\u8ba4\u503c"]]],["tbody",["tr",["td","name"],["td","Group \u4e0b\u6240\u6709 input",["span",'type="checkbox"']," \u7684 name \u5c5e\u6027"],["td","string"],["td"]],["tr",["td","value"],["td","\u5f53\u524d\u9009\u4e2d\u7684\u590d\u9009\u6846\u7684\u503c\uff0c\u53d7\u63a7\u5c5e\u6027"],["td","array"],["td"]],["tr",["td","defaultValue"],["td","\u5f53\u524d\u9009\u4e2d\u7684\u590d\u9009\u6846\u7684\u503c\uff0c\u975e\u53d7\u63a7\u5c5e\u6027"],["td","array"],["td","[]"]],["tr",["td","onChange"],["td","\u53d8\u5316\u56de\u8c03"],["td","function(e){}"],["td"]],["tr",["td","options"],["td","\u4ee5\u914d\u7f6e\u5f62\u5f0f\u8bbe\u7f6e\u5b50\u5143\u7d20"],["td","object array"],["td"]]]]],meta:{title:"Checkbox",subTitle:"\u590d\u9009\u6846",filename:"src/Checkbox/README.md"},toc:["ul",["li",["a",{className:"bisheng-toc-h1",href:"#\u590d\u9009\u6846-Checkbox",title:"\u590d\u9009\u6846 Checkbox"},"\u590d\u9009\u6846 Checkbox"]],["li",["a",{className:"bisheng-toc-h3",href:"#\u7ec4\u4ef6\u63cf\u8ff0",title:"\u7ec4\u4ef6\u63cf\u8ff0"},"\u7ec4\u4ef6\u63cf\u8ff0"]],["li",["a",{className:"bisheng-toc-h3",href:"#\u793a\u4f8b\u4ee3\u7801",title:"\u793a\u4f8b\u4ee3\u7801"},"\u793a\u4f8b\u4ee3\u7801"]],["li",["a",{className:"bisheng-toc-h2",href:"#API",title:"API"},"API"]],["li",["a",{className:"bisheng-toc-h3",href:"#Checkbox-props\u5217\u8868",title:"Checkbox props\u5217\u8868"},"Checkbox props\u5217\u8868"]],["li",["a",{className:"bisheng-toc-h3",href:"#Checkbox.Group-props\u5217\u8868",title:"Checkbox.Group props\u5217\u8868"},"Checkbox.Group props\u5217\u8868"]]]}}});