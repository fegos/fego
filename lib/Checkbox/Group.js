'use strict';

exports.__esModule = true;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _index = require('./index');

var _index2 = _interopRequireDefault(_index);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var CheckboxGroup = function (_Component) {
  _inherits(CheckboxGroup, _Component);

  function CheckboxGroup(props) {
    _classCallCheck(this, CheckboxGroup);

    var _this = _possibleConstructorReturn(this, _Component.call(this, props));

    _this.onChange = function (e) {
      var _e$target = e.target,
          value = _e$target.value,
          checked = _e$target.checked;
      var onChange = _this.props.onChange;

      var valueArr = _this.updateValueArr(value, checked);

      if (!('value' in _this.props)) {
        _this.setState({
          valueArr: valueArr
        });
      }
      onChange instanceof Function && onChange(valueArr);
    };

    _this.state = {
      valueArr: 'value' in props ? props.value : props.defaultValue
    };
    return _this;
  }

  CheckboxGroup.prototype.componentWillReceiveProps = function componentWillReceiveProps(nextProps) {
    if ('value' in nextProps && nextProps.value.toString() !== this.props.value.toString()) {
      this.setState({
        valueArr: nextProps.value
      });
    }
  };

  CheckboxGroup.prototype.updateValueArr = function updateValueArr(value, checked) {
    var valueArr = this.state.valueArr;


    if (checked) {
      valueArr.push(value);
    } else {
      var index = valueArr.indexOf(value);
      valueArr.splice(index, 1);
    }

    return valueArr;
  };

  CheckboxGroup.prototype.render = function render() {
    var _this2 = this;

    var _props = this.props,
        prefixCls = _props.prefixCls,
        className = _props.className,
        style = _props.style,
        name = _props.name,
        children = _props.children,
        options = _props.options;

    return _react2.default.createElement(
      'div',
      { className: (0, _classnames2.default)('' + prefixCls, className), style: style },
      children ? _react2.default.Children.map(children, function (child) {
        return _react2.default.cloneElement(child, {
          name: name,
          checked: _this2.state.valueArr.indexOf(child.props.value) >= 0,
          onChange: _this2.onChange
        });
      }) : options.map(function (opt) {
        return _react2.default.createElement(
          _index2.default,
          {
            key: opt.value,
            value: opt.value,
            name: name,
            checked: _this2.state.valueArr.indexOf(opt.value) >= 0,
            onChange: _this2.onChange,
            disabled: opt.disabled
          },
          opt.label
        );
      })
    );
  };

  return CheckboxGroup;
}(_react.Component);

CheckboxGroup.defaultProps = {
  // 前缀
  prefixCls: 'ns-checkbox-group',
  // 非受控属性，默认选中的值
  defaultValue: []
};
CheckboxGroup.propTypes = {
  // 前缀
  prefixCls: _propTypes2.default.oneOf(['ns-checkbox-group']),
  // checkboxGroup 下所有 input[type="checkbox"] 的 name 属性
  name: _propTypes2.default.string.isRequired,
  // 受控属性，当前选中的值
  value: _propTypes2.default.array,
  // 非受控属性，默认选中的值
  defaultValue: _propTypes2.default.array,
  // 选项变化时的回调函数
  onChange: _propTypes2.default.func,
  // 子元素
  children: _propTypes2.default.node,
  // 以配置形式设置子元素
  options: _propTypes2.default.arrayOf(_propTypes2.default.object)
};
exports.default = CheckboxGroup;