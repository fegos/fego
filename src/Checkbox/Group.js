import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import Checkbox from './index';

export default class CheckboxGroup extends Component {
  static defaultProps = {
    // 前缀
    prefixCls: 'ns-checkbox-group',
    // 非受控属性，默认选中的值
    defaultValue: [],
  }

  static propTypes = {
    // checkboxGroup 下所有 input[type="checkbox"] 的 name 属性
    name: PropTypes.string.isRequired,
    // 受控属性，当前选中的值
    value: PropTypes.array,
    // 非受控属性，默认选中的值
    defaultValue: PropTypes.array,
    // 选项变化时的回调函数
    onChange: PropTypes.func,
    // 子元素
    children: PropTypes.node,
    // 以配置形式设置子元素
    options: PropTypes.arrayOf(PropTypes.object),
  }

  constructor(props) {
    super(props);

    this.state = {
      valueArr: 'value' in props ? props.value : props.defaultValue,
    };
  }

  componentWillReceiveProps(nextProps) {
    if ('value' in nextProps && nextProps.value.toString() !== this.props.value.toString()) {
      this.setState({
        valueArr: nextProps.value,
      });
    }
  }

  updateValueArr(value, checked) {
    const { valueArr } = this.state;

    if (checked) {
      valueArr.push(value);
    } else {
      const index = valueArr.indexOf(value);
      valueArr.splice(index, 1);
    }

    return valueArr;
  }

  onChange = (e) => {
    const { value, checked } = e.target;
    const { onChange } = this.props;
    const valueArr = this.updateValueArr(value, checked);

    if (!('value' in this.props)) {
      this.setState({
        valueArr,
      });
    }
    onChange instanceof Function && onChange(valueArr);
  }

  render() {
    const {
      prefixCls, className, style, name, children, options,
    } = this.props;
    return (
      <div className={classNames(`${prefixCls}`, className)} style={style}>{
        children ? (
          React.Children.map(children, (child) => {
            return React.cloneElement(child, {
              name,
              checked: this.state.valueArr.indexOf(child.props.value) >= 0,
              onChange: this.onChange,
            });
          })
        ) : (
            options.map((opt) => {
              return (
                <Checkbox
                  key={opt.value}
                  value={opt.value}
                  name={name}
                  checked={this.state.valueArr.indexOf(opt.value) >= 0}
                  onChange={this.onChange}
                  disabled={opt.disabled}
                >
                  {opt.label}
                </Checkbox>
              );
            })
          )
      }
      </div>
    );
  }
}
