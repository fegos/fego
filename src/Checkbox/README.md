---
title: Checkbox
subTitle: 复选框
---

# 复选框 Checkbox

### 组件描述
- 同于向用户提供多选选择。

### 示例代码

```html
<Checkbox defaultChecked>默认选中的复选框</Checkbox>

<Checkbox.Group name="cbox">
  <Checkbox value="chi">语文</Checkbox>
  <Checkbox value="mat">数学</Checkbox>
  <Checkbox value="eng">英语</Checkbox>
  <Checkbox value="phy">物理</Checkbox>
  <Checkbox value="che">化学</Checkbox>
  <Checkbox value="bio">生物</Checkbox>
</Checkbox.Group>
```

## API

### Checkbox props列表

| 属性 | 说明 | 类型 | 默认值 |
|----|-----|------|------|
| label | 复选框文案 | string |  |
| children | 复选框文案，使用时label属性失效 | react node |  |
| checked | 是否选中，受控属性 | boolean |  |
| defaultChecked | 是否选中，非受控属性 | boolean |  |
| disabled | 是否禁用 | boolean | false |
| onChange | 变化回调 | function(e){} |  |
| indeterminate | 设置 indeterminate 状态，负责样式控制 | boolean | false |


### Checkbox.Group props列表

| 属性 | 说明 | 类型 | 默认值 |
|----|-----|------|------|
| name | Group 下所有 input[type="checkbox"] 的 name 属性 | string |  |
| value | 当前选中的复选框的值，受控属性 | array |  |
| defaultValue | 当前选中的复选框的值，非受控属性 | array | [] |
| onChange | 变化回调 | function(e){} |  |
| options | 以配置形式设置子元素 | object array |  |
