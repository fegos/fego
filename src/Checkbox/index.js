import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import Icon from '../Icon';
import CheckboxGroup from './Group';

export default class Checkbox extends Component {
  static defaultProps = {
    // 前缀
    prefixCls: 'ns-checkbox',
  }

  static propTypes = {
    // checkbox label 文案
    label: PropTypes.string,
    // checkbox label 文案，使用则 labe 属性失效
    children: PropTypes.node,
    // 受控属性，当前是否选中，需配合 onChane 使用
    checked: PropTypes.bool,
    // 非受控属性，当前是否选中
    defaultChecked: PropTypes.bool,
    // 是否禁用
    disabled: PropTypes.bool,
    // 变化回调函数
    onChange: PropTypes.func,
    // 设置 indeterminate 状态，负责样式控制
    indeterminate: PropTypes.bool,
  }

  constructor(props) {
    super(props);

    this.state = {
      checked: 'checked' in props ? props.checked : props.defaultChecked,
    };
  }

  componentWillReceiveProps(nextProps) {
    if ('checked' in nextProps && nextProps.checked !== this.props.checked) {
      this.setState({
        checked: nextProps.checked,
      });
    }
  }

  onChange = (e) => {
    const { onChange, disabled } = this.props;
    const { checked } = e.nativeEvent.target;

    if (disabled) return;

    if (!('checked' in this.props)) {
      this.setState({
        checked,
      });
    }

    onChange instanceof Function && onChange(e.nativeEvent);
  }

  renderLabel() {
    const { prefixCls, label, children } = this.props;

    if (children) {
      return <span className={`${prefixCls}-label`}>{children}</span>;
    } else if (label) {
      return <span className={`${prefixCls}-label`}>{label}</span>;
    }
    return null;
  }

  render() {
    const {
      prefixCls, className, style, children, indeterminate, disabled, ...otherProps
    } = this.props;

    return (
      <label htmlFor={otherProps.id} className={classNames(`${prefixCls}-wrapper`, className)} style={style}>
        <span className={classNames(`${prefixCls}`, {
          [`${prefixCls}-checked`]: this.state.checked,
          [`${prefixCls}-indeterminate`]: indeterminate,
          [`${prefixCls}-disabled`]: disabled,
        })}
        >
          <input
            type="checkbox"
            {...otherProps}
            disabled={disabled}
            className={`${prefixCls}-ipt`}
            onChange={this.onChange}
          />
          <span className={`${prefixCls}-inner`}>
            {this.state.checked ? <Icon name="yes" style={{ color: disabled ? '#aaa' : '#fff' }} /> : null}
            {(!this.state.checked && indeterminate) ? <Icon name="minus" style={{ color: '#fff' }} /> : null}
          </span>
        </span>
        {this.renderLabel()}
      </label>
    );
  }
}

Checkbox.Group = CheckboxGroup;
