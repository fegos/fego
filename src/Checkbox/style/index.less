@import "./base.less";

.@{checkbox-prefix}-wrapper {
	// 移动端点击后，eg:选中状态，点击切换至不选中状态
	// 点击后会出现 checkbox 有 hover 态时的样式
	// 所以把 hover 样式拆出来
	&:hover {
		.@{checkbox-prefix}:not(.@{checkbox-prefix}-disabled) {
			.@{checkbox-prefix}-inner {
				border-color: @primary-color;
			}
		}
	}
}