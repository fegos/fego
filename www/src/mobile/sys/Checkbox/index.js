import React, { Component } from 'react'
import { Checkbox } from 'fego'

export default class Page extends Component {
	state = { 
		checked1: true,
		checked2: false,
		checkboxGroupValue1: ['eng'],
		checkboxGroupValue2: ['phy'],
	}
	render() {
		let opt = [
			<Checkbox value='chi' key='0'>语文</Checkbox>,
			<Checkbox value='mat' key='1'>数学</Checkbox>,
			<Checkbox value='eng' key='2'>英语</Checkbox>,
			<Checkbox value='phy' key='3'>物理</Checkbox>,
			<Checkbox value='che' key='4'>化学</Checkbox>,
			<Checkbox value='bio' key='5'>生物</Checkbox>
		];
		let opts2 = [
			{ label: '语文', value: 'chi' },
			{ label: '数学', value: 'mat' },
			{ label: '英语', value: 'eng', disabled: true },
			{ label: '物理', value: 'phy'},
			{ label: '化学', value: 'che'},
			{ label: '生物', value: 'bio'},
		];
		
		return (
			<div>
				<p className='title'>单独使用的 Checkbox：</p>
				<div>
					<Checkbox defaultChecked={true} onChange={(e)=>{console.log(e.target.checked)}}>使用 defaultChecked = true</Checkbox>
				</div>
				<div>
					<Checkbox defaultChecked={false} onChange={(e)=>{console.log(e.target.checked)}}>使用 defaultChecked = false</Checkbox>
				</div>
				<div>
					<Checkbox checked={this.state.checked1} onChange={(e)=>{
						console.log(e.target.checked);
						this.setState({checked1: e.target.checked})}}>使用 checked = true</Checkbox>
				</div>
				<div>
					<Checkbox checked={this.state.checked2} onChange={(e)=>{
						console.log(e.target.checked);
						sthis.setState({checked2: e.target.checked})}}>使用 checked = false</Checkbox>
				</div>
				<div>
					<Checkbox disabled checked={true} onChange={(e)=>{console.log(e.target.checked)}} >disabled checked checkbox</Checkbox>
				</div>
				<div>
					<Checkbox disabled checked={false} onChange={(e)=>{console.log(e.target.checked)}} >disabled unchecked checkbox</Checkbox>
				</div>

				<p className='title'>组合使用的 Checkbox, 不带 defaultValue: </p>
				<Checkbox.Group name='cbox1' onChange={(v)=>{console.log(v)}}>
					{opt}
				</Checkbox.Group>

				<p className='title'>组合使用的 Checkbox, 带 defaultValue: </p>
				<Checkbox.Group name='cbox2' defaultValue={['chi']} onChange={(v)=>{console.log(v)}}>
					{opt}
				</Checkbox.Group>

				<p className='title'>组合使用的 Checkbox, 使用 value: </p>
				<Checkbox.Group name='cbox3' value={this.state.checkboxGroupValue1} onChange={(v)=>{
					console.log(v)
					this.setState({
						checkboxGroupValue1: v
					})
				}}>
					{opt}
				</Checkbox.Group>

				<p className='title'>组合使用的 Checkbox, 以配置项形式设置子元素: </p>
				<Checkbox.Group name='cbox4' options={opts2} value={this.state.checkboxGroupValue2} onChange={(v)=>{
					console.log(v)
					this.setState({
						checkboxGroupValue2: v
					})
				}} />
			</div>
		)
	}
}