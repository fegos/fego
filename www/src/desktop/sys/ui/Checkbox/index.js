import style from './index.less'
import React, { Component } from 'react'
import { Checkbox } from 'fego'

export default class Home extends Component {
	constructor(props) {
		super(props)

		this.state = {
			checked1: true,
			checked2: false,
			checkboxGroupValue1: ['eng'],
			checkboxGroupValue2: ['phy'],
		}
	}
	render() {
		let opts = [
			{ label: '语文', value: 'chi' },
			{ label: '数学', value: 'mat' },
			{ label: '英语', value: 'eng', disabled: true },
			{ label: '物理', value: 'phy'},
			{ label: '化学', value: 'che'},
			{ label: '生物', value: 'bio'},
		];
		let opts2 = [
			<Checkbox value='chi' key='0'>语文</Checkbox>,
			<Checkbox value='mat' key='1'>数学</Checkbox>,
			<Checkbox value='eng' key='2'>英语</Checkbox>,
			<Checkbox value='phy' key='3'>物理</Checkbox>,
			<Checkbox value='che' key='4'>化学</Checkbox>,
			<Checkbox value='bio' key='5'>生物</Checkbox>
		];

		return (
			<div className={style.page}>
				<div className='section'>
					<p>单独使用的 Checkbox: </p>
					<Checkbox defaultChecked={true} onChange={(e)=>{console.log(e.target.checked)}}>使用defaultChecked=true</Checkbox>
					<Checkbox defaultChecked={false} onChange={(e)=>{console.log(e.target.checked)}}>使用defaultChecked=false</Checkbox>
					<Checkbox checked={this.state.checked1} onChange={(e)=>{console.log(e.target.checked);this.setState({checked1: e.target.checked})}}>使用checked=true</Checkbox>
					<Checkbox checked={this.state.checked2} onChange={(e)=>{console.log(e.target.checked);this.setState({checked2: e.target.checked})}}>使用checked=false</Checkbox>
					<Checkbox disabled checked={true} onChange={(e)=>{console.log(e.target.checked)}} >disabled</Checkbox>
					<Checkbox disabled checked={false} onChange={(e)=>{console.log(e.target.checked)}} >disabled</Checkbox>
				</div>

				<div className='section'>
					<p>Checkbox.Group, 使用非受控属性: </p>
					<Checkbox.Group name='cbox1' onChange={(v)=>{console.log(v)}}>
						{opts2}
					</Checkbox.Group>
				</div>

				<div className='section'>
					<p>Checkbox.Group, 使用非受控属性 defaultValue: </p>
					<Checkbox.Group name='cbox2' defaultValue={['chi']} onChange={(v)=>{console.log(v)}}>
						{opts2}
					</Checkbox.Group>
				</div>

				<div className='section'>
					<p>Checkbox.Group, 使用受控属性 value: </p>
					<Checkbox.Group name='cbox3' value={this.state.checkboxGroupValue1} onChange={(v)=>{
						console.log(v)
						this.setState({
							checkboxGroupValue1: v
						})
					}}>
						{opts2}
					</Checkbox.Group>
				</div>

				<div className='section'>
					<p>Checkbox.Group, 以配置项形式设置子元素: </p>
					<Checkbox.Group name='cbox4' options={opts} value={this.state.checkboxGroupValue2} onChange={(v)=>{
						console.log(v)
						this.setState({
							checkboxGroupValue2: v
						})
					}} />
				</div>
			</div>
		)
	}
}
