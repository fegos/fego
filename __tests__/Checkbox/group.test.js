import React from 'react';
import Renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import sinon from 'sinon';
import Checkbox from 'Checkbox';

const { Group } = Checkbox;

describe('Checkbox Group test', () => {
  let opts = [];
  beforeEach(() => {
    opts = [
      <Checkbox value="chi" key="0">语文</Checkbox>,
      <Checkbox value="mat" key="1">数学</Checkbox>,
      <Checkbox value="eng" key="2">英语</Checkbox>,
      <Checkbox value="phy" key="3">物理</Checkbox>,
      <Checkbox value="che" key="4">化学</Checkbox>,
      <Checkbox value="bio" key="5">生物</Checkbox>,
    ];
  });
  it('checkbox group snapshot', () => {
    const group = Renderer.create((
      <Group name="cbox1">
        {opts}
      </Group>
    ));
    expect(group.toJSON()).toMatchSnapshot();
  });

  it('checkbox group with options should also render correctly', () => {
    const options = [
      { label: '语文', value: 'chi' },
      { label: '数学', value: 'mat' },
      { label: '英语', value: 'eng', disabled: true },
      { label: '物理', value: 'phy' },
      { label: '化学', value: 'che' },
      { label: '生物', value: 'bio' },
    ];
    const group = shallow(<Group name="cbox2" options={options} />);
    expect(group.find(Checkbox)).toHaveLength(options.length);
  });

  it('prop value changes should trigger componentWillReceiveProps method', () => {
    const group = shallow((
      <Group name="cbox3" value={['chi']}>
        {opts}
      </Group>
    ));
    const spy = sinon.spy(Group.prototype, 'componentWillReceiveProps');

    group.setProps({
      value: ['chi', 'phy'],
    });

    expect(spy.callCount).toBe(1);
  });

  it('checkbox onChange event should trigger checkboxGroup with value prop onChange callback', () => {
    const oldGroupValue = [];
    const newCheckboxValue = true;
    const spyCheckboxOnChange = sinon.spy((e) => {
      expect(spyCheckboxOnChange.callCount).toEqual(1);
      expect(e.target.checked).toEqual(newCheckboxValue);
    });
    const spyCheckboxGroupOnChange = sinon.spy(() => {
      expect(spyCheckboxGroupOnChange.callCount).toEqual(1);
    });

    const group = mount((
      <Group name="cbox4" value={oldGroupValue} onChange={spyCheckboxGroupOnChange}>
        <Checkbox onChange={spyCheckboxOnChange} value="0">checkbox</Checkbox>
      </Group>
    ));

    group.find('input').simulate('change', {
      nativeEvent: {
        target: {
          checked: newCheckboxValue,
        },
      },
    });
  });

  it('checkbox onChange event should trigger checkboxGroup with defaultValue prop onChange callback', () => {
    const oldGroupValue = [];
    const newCheckboxValue = false;
    const spyCheckboxOnChange = sinon.spy((e) => {
      expect(spyCheckboxOnChange.callCount).toEqual(1);
      expect(e.target.checked).toEqual(newCheckboxValue);
    });
    const spyCheckboxGroupOnChange = sinon.spy(() => {
      expect(spyCheckboxGroupOnChange.callCount).toEqual(1);
    });

    const group = mount((
      <Group name="cbox4" defaultValue={oldGroupValue} onChange={spyCheckboxGroupOnChange}>
        <Checkbox onChange={spyCheckboxOnChange} value="0">checkbox</Checkbox>
      </Group>
    ));

    group.find('input').simulate('change', {
      nativeEvent: {
        target: {
          checked: newCheckboxValue,
        },
      },
    });
  });
});
