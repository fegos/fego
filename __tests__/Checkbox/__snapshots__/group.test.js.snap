// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox Group test checkbox group snapshot 1`] = `
<div
  className="ns-checkbox-group"
  style={undefined}
>
  <label
    className="ns-checkbox-wrapper"
    htmlFor={undefined}
    style={undefined}
  >
    <span
      className="ns-checkbox"
    >
      <input
        checked={false}
        className="ns-checkbox-ipt"
        disabled={undefined}
        name="cbox1"
        onChange={[Function]}
        type="checkbox"
        value="chi"
      />
      <span
        className="ns-checkbox-inner"
      />
    </span>
    <span
      className="ns-checkbox-label"
    >
      语文
    </span>
  </label>
  <label
    className="ns-checkbox-wrapper"
    htmlFor={undefined}
    style={undefined}
  >
    <span
      className="ns-checkbox"
    >
      <input
        checked={false}
        className="ns-checkbox-ipt"
        disabled={undefined}
        name="cbox1"
        onChange={[Function]}
        type="checkbox"
        value="mat"
      />
      <span
        className="ns-checkbox-inner"
      />
    </span>
    <span
      className="ns-checkbox-label"
    >
      数学
    </span>
  </label>
  <label
    className="ns-checkbox-wrapper"
    htmlFor={undefined}
    style={undefined}
  >
    <span
      className="ns-checkbox"
    >
      <input
        checked={false}
        className="ns-checkbox-ipt"
        disabled={undefined}
        name="cbox1"
        onChange={[Function]}
        type="checkbox"
        value="eng"
      />
      <span
        className="ns-checkbox-inner"
      />
    </span>
    <span
      className="ns-checkbox-label"
    >
      英语
    </span>
  </label>
  <label
    className="ns-checkbox-wrapper"
    htmlFor={undefined}
    style={undefined}
  >
    <span
      className="ns-checkbox"
    >
      <input
        checked={false}
        className="ns-checkbox-ipt"
        disabled={undefined}
        name="cbox1"
        onChange={[Function]}
        type="checkbox"
        value="phy"
      />
      <span
        className="ns-checkbox-inner"
      />
    </span>
    <span
      className="ns-checkbox-label"
    >
      物理
    </span>
  </label>
  <label
    className="ns-checkbox-wrapper"
    htmlFor={undefined}
    style={undefined}
  >
    <span
      className="ns-checkbox"
    >
      <input
        checked={false}
        className="ns-checkbox-ipt"
        disabled={undefined}
        name="cbox1"
        onChange={[Function]}
        type="checkbox"
        value="che"
      />
      <span
        className="ns-checkbox-inner"
      />
    </span>
    <span
      className="ns-checkbox-label"
    >
      化学
    </span>
  </label>
  <label
    className="ns-checkbox-wrapper"
    htmlFor={undefined}
    style={undefined}
  >
    <span
      className="ns-checkbox"
    >
      <input
        checked={false}
        className="ns-checkbox-ipt"
        disabled={undefined}
        name="cbox1"
        onChange={[Function]}
        type="checkbox"
        value="bio"
      />
      <span
        className="ns-checkbox-inner"
      />
    </span>
    <span
      className="ns-checkbox-label"
    >
      生物
    </span>
  </label>
</div>
`;
